server/src/main/java/org/elasticsearch/search/fetch/subphase/FieldFetcher.java

```diff
@@ -54,16 +71,27 @@ public static FieldFetcher create(SearchExecutionContext context, Collection<Fie
 
             for (String field : context.getMatchingFieldNames(fieldPattern)) {
                 MappedFieldType ft = context.getFieldType(field);
+                if (ft == null) { // unmapped field
```

Comment:
why is this new conditional required only now?

Reply:
I saw failures for this already before in some local tests and I am not sure why I didn't see them in the CI...I guess again some tests are not executed if there are other tests failing. This happens for some fields

Reply:
That chack is required because later on we do `ft.isStored()` which we need to do because we need to check if we need to fetch fields that are part of `stored_fields`

Reply:
Notice that the behavior of `FecthFields` or put it another way, the way of handling `fields` is described by the test SizeMappingIT like

```
assertResponse(
            prepareSearch("test").addFetchField("*"),
            response -> assertNull(response.getHits().getHits()[0].getFields().get("_size"))
        );
```

which is...if a wildcard is used we **do not return** metadata fields such as `_size`.

Reply:
Come se non bastassea quando i field sono richiesti attraverso `stored_fields` la chiamata a `context.getFieldType(field)` restituisce null come `MappedFieldType`. Questo pero' non e' un problema perche' in quel caso quei field saranno recuperati attraverso la `StoredFieldPhase` e quindi il `ValueFecther` non verra' usato...in effetti non possiamo usarlo in quel caso perche' `MappedFiledType` e' **null**.

Reply:
In summary:
1. We need to check for `MappedFieldType` to be `null` because when handling fields passed as `stored_fields` in the `StoredFieldContext` we need to skip **some of them**...and we can't do otherwise because in that case `MappedFieldType` is null.
2. We can't add a ft = null to the list of resolved fields later on in the loop because otherwise later on sorting of resolved fields breaks with a NPE. See `resolvedFields.sort(Comparator.comparing(resolvedField -> resolvedField.mappedFieldType.name()));`

Reply:
I honestly don't know why `context.getFieldType(field)` is null when handling stored fields...maybe there is no MapperService available?